### PR TITLE
Implement `.engine-release.version` files for engine Skia Gold tests

### PR DESCRIPTION
**Pull request by matanlurey**
Work towards https://github.com/flutter/flutter/issues/144835.

Doc (_sorry, internal only_): [go/flutter-engine-goldens-workflow](http://goto.google.com/flutter-engine-goldens-workflow).

This implements the majority of the proposed workflow, that is, optionally having a plain-text version at the root of the directory, and using it to apply a unique suffix we can review in release branches. As it stands, this is a NO-OP outside of tests (it will have no impact, and can be ignored).

What's missing before using this feature in release branches:

- Optimization work with the infra team (not sure if blocking or not):
  https://github.com/flutter/flutter/issues/145842
- A dry-run of this with the release team to make sure it works as intended

@gaaclarke As implemented, I _think_ we don't need anything special for [`dir_contents_diff`](https://github.com/flutter/engine/blob/b0d3663f439fd97c32bf4d46b9004c97841c43b8/tools/dir_contents_diff), but maybe I'm wrong - I think only the _test_ names are being changed, not the names on disk.

/cc @zanderso as well.

### CODE REVIEW

File: .engine-release.version
Comment (matanlurey) [inline .engine-release.version:1]: Oops, the comments are out of date here. Will fix tomorrow.

**Review by zanderso — APPROVED** [review]

**Comment by matanlurey** [comment]
Thanks! Friendly ping @mdebbar @gaaclarke. I promise 90% of the diff is comments/docs/tests.

File: testing/skia_gold_client/README.md [hunk 1]
```diff
@@ -1,70 +1,111 @@
...
+   ```diff
+   # This is just an example.
+   targets:
+     - name: Linux linux_android_emulator_tests
+       properties:
+         config_name: linux_android_emulator
+         + dependencies: >-
+         +   [
+         +     {"dependency": "goldctl", "version": "git_revision:720a542f6fe4f92922c3b8f0fdcc4d2ac6bb83cd"}
+         +   ]
```
Comment (gaaclarke) [inline testing/skia_gold_client/README.md:69]: Does this render correctly?  I would have expected the pluses and minuses at the beginning of the line

File: testing/skia_gold_client/lib/src/release_version.dart [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  ///
+  /// 3.21
+  /// ```
+  ///
+  /// ... would parse to `ReleaseVersion(major: 3, minior: 21)`.
+  ///
+  /// Throws a [FormatException] if the file contents are not in the expected
+  /// format (either empty/comments only, or missing a `major.minor` line in
+  /// the format described in [ReleaseVersion.new]).
+  static ReleaseVersion? parse(String fileContents) {
```
Comment (gaaclarke) [inline testing/skia_gold_client/lib/src/release_version.dart:55]: nit: I think this could be 1/3 the size if you use regular expressions.

**Review by gaaclarke — APPROVED** [review]
LGTM.  Sorry, I passed over this when I saw it was already approved.  That's for pinging me again.

File: testing/skia_gold_client/README.md (hunk 1 above)
Comment (matanlurey) [reply testing/skia_gold_client/README.md:69]: Hmm good point, let me try again

File: testing/skia_gold_client/lib/src/release_version.dart (hunk 2 above)
Comment (matanlurey) [reply testing/skia_gold_client/lib/src/release_version.dart:55]: Hmm. I hope this is fine, I am not opposed to a regular expression but also hate multi-line regular expressions personally.

**Comment by matanlurey** [comment]
Thanks! @godofredoc friendly ping as well - I'm not sure you want to review this or not, but I figured I'd give you a chance before merging.

**Review by mdebbar — APPROVED** [review]
LGTM!
